tests/explicit-nulls/warn/i21577.scala

```diff
@@ -0,0 +1,24 @@
+def f(s: String) =
+  val s2 = s.trim()
+  s2 match
+    case s3: String => println(1)
+    case _ => println(2) // warn
+
+
+def f2(s: String | Null) =
+  val s2 = s.nn.trim()
+    s2 match
```

Comment:
I guess `f` and `f2` are testing the same thing? Both selectors are flexible types.